### PR TITLE
PLATFORM-4509 | Add controller for MultiDelete

### PR DESCRIPTION
**Pull request by maaartyyynaa**

### CODE REVIEW

File: extensions/wikia/MultiTasks/MultiDeleteController.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+		foreach ( $pagesToDelete as $pageName ) {
+			$page = WikiPage::factory( \Title::newFromText( $pageName ) );
+
+			if ( $page !== null ) {
+				$e = '';
+				$page->doDeleteArticle( $reason, false, null, null, $e, $user );
+			}
+		}
+		$this->response->setCode( 200 );
```
Comment (themech) [inline extensions/wikia/MultiTasks/MultiDeleteController.php]:
```suggestion
		$this->response->setCode( WikiaResponse ::RESPONSE_CODE_OK );
```

File: extensions/wikia/MultiTasks/MultiDeleteController.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		return;
```
Comment (themech) [inline extensions/wikia/MultiTasks/MultiDeleteController.php]: Not needed here
```suggestion
```

File: extensions/wikia/MultiTasks/MultiDeleteController.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+			return;
+		}
+		$user = User::newFromId( $userId );
+
+		foreach ( $pagesToDelete as $pageName ) {
+			$page = WikiPage::factory( \Title::newFromText( $pageName ) );
+
+			if ( $page !== null ) {
+				$e = '';
+				$page->doDeleteArticle( $reason, false, null, null, $e, $user );
```
Comment (themech) [inline extensions/wikia/MultiTasks/MultiDeleteController.php]: Two issues here:
* `null` is passed as article id, mixing up types; this should be 0
* also the boolean `$commit` value is set to null which is wrong for two reasons; first it is also mixing up the types; second, passing an empty value will probably start a new transaction but will not commit it, so the article won't be deleted

File: extensions/wikia/MultiTasks/MultiDeleteController.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		if ( empty( $pagesToDelete ) || empty( $userId ) ||  empty( $reason ) ) {
+			$this->response->setCode( WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+			return;
+		}
+		$user = User::newFromId( $userId );
+
+		foreach ( $pagesToDelete as $pageName ) {
+			$page = WikiPage::factory( \Title::newFromText( $pageName ) );
+
+			if ( $page !== null ) {
```
Comment (themech) [inline extensions/wikia/MultiTasks/MultiDeleteController.php:33]: Let's make use of the `Loggable` trait and log information about missing pages :)

**Comment by maaartyyynaa** [comment]
Closing.

**Pull request closed without merging**